docs/component-specs.md

```diff
@@ -1,12 +1,65 @@
 # Spec files for agent-managed components
 
-Spec files are YML files that describe the capabilities of a component and when / how to run it. They are used by Agent to convert a user-provided policy into a configured running process.
+Spec files are YML files that describe the capabilities of a component and when / how to run it. They are used by Agent to convert a user-provided [policy](agent-policy.md) into a configured running process.
 
-[This documentation should be expanded](https://github.com/elastic/elastic-agent/issues/2690)
+## Spec file layout
 
-## Preventions
+A spec file corresponds to a specific executable, which must be in the same directory and have the same name -- for example `filebeat.spec.yml` would correspond to the executable `filebeat`, or `filebeat.exe` on Windows. The configuration is broken into sections:
 
-Components may include a `runtime.preventions` section containing [EQL conditions](https://www.elastic.co/guide/en/elasticsearch/reference/current/eql-syntax.html#eql-syntax-conditions) which should prevent the use of that component if any are true. Each prevention should include a `condition` in EQL syntax and a `message` that will be displayed if the condition prevents the use of a component.
+```yml
+version: 2
+inputs:
+  - name: <input name 1>
+    ...
+  - name: <input name 2>
+    ...
+shippers:
+  - name: <shipper name 1>
+    ...
+```
+
+The `version` key must be present and must equal 2 (to distinguish from the older version 1 schema that is no longer supported).
+
+`inputs` is a list of input types this component can run, and `shippers` is a list of shipper types this component can run.
```

Comment:
The `outputs` section should also be mentioned here.

Reply:
`outputs` isn't its own section in the spec file as far as I understand. It's just a list of strings on individual `input` and `shipper` configs, and is documented in that section below. I added a bit more to this paragraph to clarify this, let me know if there was something more you intended.

Reply:
Right, I've it is per input/shipper. Resolving.